### PR TITLE
Update several py3 packages following dependabot alerts

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please abort
Let's also update pydantic [link](https://github.com/cms-sw/cmsdist/security/dependabot/310)

**Comment by iarspider** [comment]
please test

File: gosam.spec [hunk 1]
```diff
@@ -12,6 +13,7 @@ Requires: python3 py3-cython
 %prep
 %setup -q -n %{n}-%{realversion}
 sed -i -e 's|^VERSION =.*|VERSION = "%{realversion}"|;s|^GIT_REVISION =.*|GIT_REVISION = "%{tag}"|' setup.py
+%patch0 -p1
```
Comment (smuzaffar) [inline gosam.spec]: @iarspider , why was this change needed?

Comment (iarspider) [reply gosam.spec]: @smuzaffar `setuptools` complains that `2.1.0-2.1.0` is a malformed verison

Comment (smuzaffar) [reply gosam.spec]: ah ok @iarspider . I guess there is no point having an extra patch for `setup.py` and also using `sed` to update it.   I would suggest to just update the `sed -i -e ` command above and fix the  `TAR_VERSION` too e.g.
```
sed -ir -e 's#^((VERSION|GIT_REVISION|TAR_VERSION) *= *).*#\1"%{realversion}"#' setup.py
```

also add the following before the `sed` command
```
grep -q '^VERSION *=' setup.py && grep -q '^GIT_REVISION *=' setup.py && grep -q '^TAR_VERSION *=' setup.py
```

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
hold
updating py3-setuptools/py3-pip should be delayed until we need to rebuild the entire stack

**Comment by smuzaffar** [comment]
@iarspider , can you please revert the pip and setuptools change.

**Comment by iarspider** [comment]
No, since we need updated setuptools and pip for onnx update

**Comment by smuzaffar** [comment]
> No, since we need updated setuptools and pip for onnx update

ah ok. Then can you please run `cms-bot/utils/cmsdist_pip_pkgs_update.py -u -c cmsdist` and see what other pythn packages we can update along with this. Note that `cmsdist_pip_pkgs_update.py -u`  automatically check for updates and changes cmsdist/pip/requirements.txt file but you need to locally build every thing to make sure all the updated packages go along nicely . I would suggest to update those packages which do not require a lot changes e.g simple version update without any extra deps etc.

**Comment by iarspider** [comment]
please test

**Comment by smuzaffar** [comment]
please test with #9166

**Comment by smuzaffar** [comment]
please test with https://github.com/cms-sw/cmsdist/pull/9166 for el8_aarch64_gcc12

**Comment by smuzaffar** [comment]
please test with https://github.com/cms-sw/cmsdist/pull/9166 for el9_amd64_gcc12

**Comment by iarspider** [comment]
please test for el9_amd64_gcc12

**Comment by iarspider** [comment]
please abort

**Comment by iarspider** [comment]
please test for el9_amd64_gcc12

**Comment by iarspider** [comment]
please test for el8_aarch64_gcc12

**Comment by iarspider** [comment]
please test for el9_amd64_gcc12

**Comment by iarspider** [comment]
please test for el9_amd64_gcc12

**Comment by iarspider** [comment]
please test for el9_amd64_gcc12

**Comment by iarspider** [comment]
please abort

**Comment by iarspider** [comment]
please test for el9_amd64_gcc12

**Comment by smuzaffar** [comment]
test parameters:
- pull_request = #9166

**Comment by smuzaffar** [comment]
please test for el9_amd64_gcc12

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test for slc7_amd64_gcc12

**Comment by iarspider** [comment]
please test for el9_amd64_gcc12

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test for slc7_amd64_gcc12

**Comment by smuzaffar** [comment]
@iarspider , TF for slc7 failed to build with error
```
ImportError: urllib3 v2 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'OpenSSL 1.0.2k-fips  26 Jan 2017'. See: https://github.com/urllib3/urllib3
/issues/2168
```
looks like we need to revert urllib3 or add two version one for slc7 and one for other ( just like we do for `crytography`). If you update the PR before the tests for el8/el9 are done then please only start tests for slc7 otherwise running el8/el9 tests will be killed

**Comment by smuzaffar** [comment]
please test for slc7_amd64_gcc12

lets start tests for slc7 only (el8/el9 tests are already runnning)
@iarspider , I have updated requirements.txt to use old urllib3 for slc7

**Comment by smuzaffar** [comment]
el9: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4b5e43/39282/summary.html
el8: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4b5e43/39283/summary.html

@iarspider , there are some relocation/unknown file failures e.g. https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4b5e43/39283/external_checks/relocate/ and https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4b5e43/39283/external_checks/unknown/ . Once this ismerged then please open a PR to fix this as soon as possible

**Comment by smuzaffar** [comment]
+externals

el8/el9/slc7 tests looks good. lets get this in for nextIB
